tests/test_geometry_3d.h

```diff
@@ -0,0 +1,417 @@
+				radius(p_radius), height(p_height), sides(p_sides), lats(p_lats), axis(p_axis), want_size(p_want){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(10, 20, 6, 10, Vector3::Axis(), 126));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case current_case = tt[i];
+		Vector<Plane> capsule = Geometry3D::build_capsule_planes(current_case.radius, current_case.height, current_case.sides, current_case.lats, current_case.axis);
+		// Should equal (p_sides * p_lats) * 2 + p_sides
+		CHECK(capsule.size() == current_case.want_size);
+	}
+}
+TEST_CASE("[Geometry3D] Build Cylinder Planes") {
+	struct Case {
+		real_t radius, height;
+		int sides;
+		Vector3::Axis axis;
+		int want_size;
+		Case(){};
+		Case(real_t p_radius, real_t p_height, int p_sides, Vector3::Axis p_axis, int p_want) :
+				radius(p_radius), height(p_height), sides(p_sides), axis(p_axis), want_size(p_want){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(3.0f, 10.0f, 10, Vector3::Axis(), 12));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case current_case = tt[i];
+		Vector<Plane> planes = Geometry3D::build_cylinder_planes(current_case.radius, current_case.height, current_case.sides, current_case.axis);
+		CHECK(planes.size() == current_case.want_size);
+	}
+}
+TEST_CASE("[Geometry3D] Build Sphere Planes") {
+	struct Case {
+		real_t radius;
+		int lats, lons;
+		Vector3::Axis axis;
+		int want_size;
+		Case(){};
+		Case(real_t p_radius, int p_lat, int p_lons, Vector3::Axis p_axis, int p_want) :
+				radius(p_radius), lats(p_lat), lons(p_lons), axis(p_axis), want_size(p_want){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(10.0f, 10, 3, Vector3::Axis(), 63));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case current_case = tt[i];
+		Vector<Plane> planes = Geometry3D::build_sphere_planes(current_case.radius, current_case.lats, current_case.lons, current_case.axis);
+		CHECK(planes.size() == 63);
+	}
+}
+TEST_CASE("[Geometry3D] Build Convex Mesh") {
+	struct Case {
+		Vector<Plane> object;
+		int want_faces, want_edges, want_vertices;
+		Case(){};
+		Case(Vector<Plane> p_object, int p_want_faces, int p_want_edges, int p_want_vertices) :
+				object(p_object), want_faces(p_want_faces), want_edges(p_want_edges), want_vertices(p_want_vertices){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(Geometry3D::build_box_planes(Vector3(5, 10, 5)), 6, 12, 8));
+	tt.push_back(Case(Geometry3D::build_capsule_planes(5, 5, 20, 20, Vector3::Axis()), 820, 7603, 6243));
+	tt.push_back(Case(Geometry3D::build_cylinder_planes(5, 5, 20, Vector3::Axis()), 22, 100, 80));
+	tt.push_back(Case(Geometry3D::build_sphere_planes(5, 5, 20), 220, 1011, 522));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case current_case = tt[i];
+		Geometry3D::MeshData mesh = Geometry3D::build_convex_mesh(current_case.object);
+		CHECK(mesh.faces.size() == current_case.want_faces);
+		CHECK(mesh.edges.size() == current_case.want_edges);
+		CHECK(mesh.vertices.size() == current_case.want_vertices);
+	}
+}
```

Comment:
@Gorgonx7 Can you help me debug an issue with this? The problem I'm having is that this test case seems to be very fragile. Any slight change causes the test to fail.

In #37842, the constructor for Plane with normal and point had its arguments swapped, so we took the opportunity to improve the code in `Geometry3D::build_sphere_planes` by just passing in the radius distance directly instead of constructing using a point and normal (which the constructor just turns back into a distance).

Here are some examples of the planes being passed. Before:

```
[N: (0.791327, -0.257118, 0.554700), D: 5.000001]
[N: (0.527551, -0.171412, 0.832050), D: 5.000001]
[N: (0.230665, -0.074948, 0.970142), D: 4.999999]
```

After:

```
[N: (0.791327, -0.257118, 0.554700), D: 5]
[N: (0.527551, -0.171412, 0.832050), D: 5]
[N: (0.230665, -0.074948, 0.970142), D: 5]
```

The after output is slightly more precise, but it breaks the test case code:

```
tests/test_geometry_3d.h:154:
TEST CASE:  [Geometry3D] Build Convex Mesh

tests/test_geometry_3d.h:171: ERROR: CHECK( mesh.edges.size() == current_case.want_edges ) is NOT correct!
  values: CHECK( 1009 == 1011 )
```

Reply:
@aaronfranke sorry I never got back to you, things have been a bit hectic at work, so hectic I've not had much luck working on open source at the moment, did you ever figure this out?

Reply:
@Gorgonx7 No, currently the test is just disabled in master.